### PR TITLE
Add namespace to serviceCluster parameter used by Envoy as the servic…

### PR DESCRIPTION
**Pull request by objectiser**
…e name for span data

Currently span data reported by the istio proxy/envoy uses the "app" value as the service name.

The problem is that it is possible the same "app" name could be used in multiple namespaces, and therefore it is not clear which app is actually reporting the data.

One solution would be to report the namespace as a tag on the span data, to enable it to be distinguished. However the benefit of including the namespace as part of the service name is that it is clear that the apps are different - for example on a dependency graph.

The following images show how the service name (using "app:namespace" format) looks in the jaeger trace view and dependency graph (bookinfo deployed in `test` namespace):

istio-jaeger-namespace-test-trace
istio-jaeger-namespace-test-dag

### CODE REVIEW

**Comment by douglas-reid** [comment]
@objectiser apologies. i was ooo when this was created and missed the notifications. in general, this seems like a reasonable approach. curious: why `:` instead of `.` as the delimiter between name and namespace?

**Comment by objectiser** [comment]
@douglas-reid np - main reason was to pick something that was unlikely to be used in either the namespace or name - but have no strong opinion if would prefer '.' or any other char.

**Comment by douglas-reid** [comment]
'.' would be more consistent with other parts of the system, from my perspective.

**Comment by objectiser** [comment]
@douglas-reid I'll make the change early next week.

**Comment by objectiser** [comment]
@douglas-reid updated.

**Comment by douglas-reid** [comment]
@objectiser this LGTM. should we get someone more familiar with pilot to take a second look?

**Comment by objectiser** [comment]
Sure. Who would you suggest?

File: istioctl/cmd/istioctl/testdata/v1alpha3/merged-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
```
Comment (rshriram) [inline istioctl/cmd/istioctl/testdata/v1alpha3/merged-gateway.yaml]: ? why these files?

**Comment by abbi-gaurav** [comment]
@rshriram Would it be possible to know if this already out in a release or slated for one?

**Pull request merged**